src/main/java/org/broadinstitute/hellbender/tools/walkers/gnarlyGenotyper/GnarlyGenotyperEngine.java

```diff
@@ -267,10 +272,10 @@ private GenotypesContext iterateOnGenotypes(final VariantContext vc, final List<
         if (!summarizePLs) {
             final int maximumAlleleCount = inputAllelesWithNonRef.size();
             final int numConcreteAlts = maximumAlleleCount - 2; //-1 for NON_REF and -1 for ref
-            if (maximumAlleleCount <= maxAltAllelesToOutput) {
-                newPLsize = likelihoodSizeCache[numConcreteAlts]; //-1 for zero-indexed array
+            if (maximumAlleleCount <= maxAltAllelesToOutput + 1) {
+                newPLsize = likelihoodSizeCache[numConcreteAlts + 1]; //cache is indexed by #alleles with ref; don't count NON_REF
```

Comment:
Are you happy with these lingering `+ 1`'s @ldgauthier ? Perhaps we should just declare descriptively-named variables with the values we want (eg., `final int maxAllelesToOutput = maxAltAllelesToOutput + 1`) to cut down on the possibility of confusion in the distant future?